src/test/java/io/nats/compatibility/ObjectStoreCommand.java

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    private void doPutLink() {
+        try {
+            String linkName = JsonValueUtils.readString(full, "link_name");
+
+            ObjectStore os = nc.objectStore(bucket);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ObjectInfo oi = os.get(object, baos);
+            os.addLink(linkName, oi);
+            respond();
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doWatch() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchWatcher ww = new WatchWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(ww)) {
+                ww.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(ww.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = result + "," + oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    private void doWatchUpdate() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchUpdatesWatcher wuw = new WatchUpdatesWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(wuw)) {
+                wuw.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(wuw.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchUpdatesWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    protected void _respondDigest(ObjectInfo oi) {
+        String digest = oi.getDigest();
+        Log.info("RESPOND " + subject + " digest " + digest);
+        byte[] payload = Base64.getUrlDecoder().decode(digest.substring(8));
+        nc.publish(replyTo, payload);
+    }
+
+    private void _putObject(String objectName, String description) {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            ObjectMeta meta = ObjectMeta.builder(objectName).description(description).build();
+            InputStream inputStream = Utility.getFileAsInputStream("nats-server.zip");
```

Comment:
I don't think you should keep `nats-server.zip` in the reposiotry - compat tester gives you the URL to the thing you should put in the object store. You can simply download whatever's in the received config and use this instead. That way you are not risking having the test fails if at some point the url in config changes (for whatever reason).

Reply:
What's the difference between that and hard-coding bucket names in the default test?

Reply:
Also, why is there a url in the get-link test?

Reply:
Changed code to download the file from the url in the json

Reply:
I think the url in get link is there by accident, it has no value for the client.